toolchain/sem_ir/typed_insts.h

```diff
@@ -881,6 +885,28 @@ struct ReturnExpr {
   InstId dest_id;
 };
 
+// The return slot of a function declaration, as exposed in the function body.
+// This acts as an output parameter, analogous to `BindName` for input
+// parameters.
+struct ReturnSlot {
+  static constexpr auto Kind =
+      InstKind::ReturnSlot.Define<Parse::NodeId>({.ir_name = "return_slot"});
+
+  TypeId type_id;
+  InstId value_id;
```

Comment:
OK, how's this?

> (though it might be good to separately associate the specific instruction representing that expression in the source for diagnostics)

Do you see any places we can use that right now? If not, I'd rather wait until we have a need for it.